cpufreq/src/cpufreq-applet.c

```diff
@@ -130,563 +130,566 @@ G_DEFINE_TYPE (CPUFreqApplet, cpufreq_applet, PANEL_TYPE_APPLET)
-
-                text_perc = g_strdup_printf ("%d%%", perc);
-                gtk_label_set_text (GTK_LABEL (applet->label), text_perc);
-                g_free (text_perc);
-        }
-
-        if (applet->show_unit) {
-                gtk_label_set_text (GTK_LABEL (applet->unit_label), unit_label);
-                /*Hold the largest size set by MHZ or GHZ to prevent jumping */
-                gtk_widget_get_preferred_size (GTK_WIDGET (applet->unit_label),&req, NULL);
-                gtk_widget_set_size_request (GTK_WIDGET (applet->unit_label),req.width, req.height);
-        }
-
-        if (applet->show_icon) {
-                cpufreq_applet_pixmap_set_image (applet, perc);
-        }
-
-	if (governor) {
-		gchar *gov_text;
-
-		gov_text = g_strdup (governor);
-		gov_text[0] = g_ascii_toupper (gov_text[0]);
-		text_mode = g_strdup_printf ("%s\n%s %s (%d%%)",
-					     gov_text, freq_label,
-					     unit_label, perc);
-		g_free (gov_text);
-	}
-
-        g_free (freq_label);
-        g_free (unit_label);
-
-	if (text_mode) {
-		gchar *text_tip;
-
-		text_tip = cpufreq_utils_get_n_cpus () == 1 ?
-			g_strdup_printf ("%s", text_mode) :
-			g_strdup_printf ("CPU %u - %s", cpu, text_mode);
-		g_free (text_mode);
-
-		gtk_widget_set_tooltip_text (GTK_WIDGET (applet), text_tip);
-		g_free (text_tip);
-	}
-
-        /* Call refresh only the first time */
-        if (applet->need_refresh) {
-                cpufreq_applet_refresh (applet);
-		applet->need_refresh = FALSE;
+   gchar          *text_mode = NULL;
+   gchar          *freq_label, *unit_label;
+   gint            freq;
+   gint            perc;
+   guint           cpu;
+   GtkRequisition  req;
+   const gchar    *governor;
+
+   cpu = cpufreq_monitor_get_cpu (monitor);
+   freq = cpufreq_monitor_get_frequency (monitor);
+   perc = cpufreq_monitor_get_percentage (monitor);
+   governor = cpufreq_monitor_get_governor (monitor);
+
+   freq_label = cpufreq_utils_get_frequency_label (freq);
+   unit_label = cpufreq_utils_get_frequency_unit (freq);
+
+   if (applet->show_freq) {
+       /*Force the label to render if frequencies are not found right away*/
+       if (freq_label == NULL){
+           gtk_label_set_text (GTK_LABEL (applet->label),"---");
+       }
+        else {
+           gtk_label_set_text (GTK_LABEL (applet->label), freq_label);
         }
+        /*Hold the largest size set by any jumping text */
+        gtk_widget_get_preferred_size (GTK_WIDGET (applet->label),&req, NULL);
+        gtk_widget_set_size_request (GTK_WIDGET (applet->label),req.width, req.height);
+    }
+
+    if (applet->show_perc) {
+        gchar *text_perc;
+
+        text_perc = g_strdup_printf ("%d%%", perc);
+        gtk_label_set_text (GTK_LABEL (applet->label), text_perc);
+        g_free (text_perc);
+    }
+
+    if (applet->show_unit) {
+        gtk_label_set_text (GTK_LABEL (applet->unit_label), unit_label);
+        /*Hold the largest size set by MHZ or GHZ to prevent jumping */
```

Comment:
```diff
-        /*Hold the largest size set by MHZ or GHZ to prevent jumping */
+        /* Hold the largest size set by MHZ or GHZ to prevent jumping */
```